client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-plugin-install/index.tsx

```diff
@@ -0,0 +1,79 @@
+import { StepContainer } from '@automattic/onboarding';
+import { useI18n } from '@wordpress/react-i18n';
+import { useEffect } from 'react';
+import wpcomRequest from 'wpcom-proxy-request';
+import DocumentHead from 'calypso/components/data/document-head';
+import { LoadingEllipsis } from 'calypso/components/loading-ellipsis';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import type { Step } from '../../types';
+
+const SiteMigrationPluginInstall: Step = function WaitForAtomic( { navigation, data } ) {
+	const { submit } = navigation;
+	const siteId = data?.siteId;
+	const siteSlug = data?.siteSlug;
+	const { __ } = useI18n();
+
+	useEffect( () => {
+		if ( ! siteId || ! siteSlug ) {
+			return;
+		}
+
+		const installSiteMigrationPlugins = async () => {
```

Comment:
can we move this function to outside the use effect contexts?
If we keep it here, it will be relocated memory in every single frame rendering.

Reply:
Our eslint rules say that moving the function outside of the `useEffect` will cause re-renders every time.